quipucords/api/common/entities.py

```diff
@@ -154,6 +155,33 @@ def provider_type(self):
             return value
         return None
 
+    @property
+    def role(self):
+        """Retrieve system purpose role."""
+        role = self._fingerprints.system_role
+        list_roles = set(item.value for item in SystemPurposeRole)
+        if role in list_roles:
+            return role
+        return None
```

Comment:
FYI, since this is a set, not a list, I'd call the variable accordingly.
Also, since this is a set comprehension you could drop the `set` and just use curly braces

`        set_roles = {item.value for item in SystemPurposeRole}`

but I'd rather propose something else. only loop through SystemPurposeRole until we find the value.
best case scenario we hit the result right away

```
        for system_purpose_role in SystemPurposeRole:
            if role == system_purpose_role.value:
                return role
        return None
```

buuuuuuuuuut, there's something else bothering me here: these new properties are starting to pollute HostEntity. It's convenient they are called role/sla/usage but these names only make sense under `system_purpose` namespace.

we could either prefix all these variables with `system_purpose_` to make it clear OR maybe just build the whole object in a single method

```python
    def system_purpose(self):
        """Return system_purpose dictionary."""
        system_purpose_map = (
            ("role", self._fingerprints.system_role, SystemPurposeRole),
            ("usage", self._fingerprints.system_usage_type, SystemPurposeUsage),
            ("sla", self._fingerprints.system_service_level_agreement, SystemPurposeSla),
        )
        _system_purpose = {}
        for name, attr, enum in system_purpose_map:
            # since inventory expects specific values for each key in system_purpose dict, we need to check their values
            # ref: https://github.com/RedHatInsights/insights-host-inventory/blob/b2c067f/swagger/system_profile.spec.yaml#L526-L547
            for enum_item in enum:
                if attr == enum_item.value:
                    _system_purpose[name] = attr
                    # since we found an appropriate value, skip looping
                    break
        return _system_purpose
```

if you follow this last suggestion, remember to remove 'source=*' from `SystemPurposeSerializer` invocation

Reply:
I went for your suggestion. Thanks!